packages/core/crates/core-misc/src/environment.rs

```diff
@@ -0,0 +1,265 @@
+use real_base::real;
+use serde::{Deserialize, Serialize};
+use wasm_bindgen::prelude::*;
+use wasm_bindgen::JsValue;
+
+#[wasm_bindgen(module = "@hoprnet/hopr-real")]
+extern "C" {
+    #[wasm_bindgen(catch)]
+    pub fn coerce_version(version: String) -> Result<String, JsValue>;
+
+    #[wasm_bindgen(catch)]
+    pub fn satisfies(version: String, range: String) -> Result<bool, JsValue>;
+}
+
+#[derive(Deserialize, Serialize, Clone, Copy)]
+#[serde(rename_all(deserialize = "lowercase"))]
+pub enum EnvironmentType {
+    Production,
+    Staging,
+    Development,
+}
+
+impl ToString for EnvironmentType {
+    fn to_string(&self) -> String {
+        match self {
+            Self::Production => String::from("production"),
```

Comment:
Idiomatic way is `"my fancy string".into()`

That applies to the most situations where you'd use `X::from(y)` and compiler knows the type of the receiving side.

Reply:
changed at all easy occurences